File: 백준/Bronze/1110. 더하기 사이클/더하기 사이클.cc
#include <stdio.h> 
#include <string.h>
#include <stdlib.h>
 


int main()
{
	int N=0,a=0,b=0,res=0,tmp=0,cnt=0;
	
	scanf("%d",&N);
	tmp=N;
	
	while(1){
		if(N<10){
			a=0;
			b=N;
		}
		else{
			b=N%10;
			a=N/10;
		}
		res=a+b;
		N=b*10+res%10;
		cnt++;
		if(N==tmp){
			printf("%d",cnt);
			break;
		}
	}
	
	return 0;
}

File: 백준/Bronze/1110. 더하기 사이클/더하기 사이클.py
import sys
import math
input=sys.stdin.readline

N=int(input())
init=N
cnt=0
while(1):
    tmp=N//10+N%10
    N=N%10*10+tmp%10
    cnt+=1
    if(N==init):
        print(cnt)
        break